MultipleFileUpload/Model/MySQL/Queue.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * This file is part of the MultipleFileUpload (https://github.com/jkuchar/MultipleFileUpload/)
+ *
+ * Copyright (c) 2014 Ciki (https://github.com/Ciki)
+ * Copyright (c) 2014 Jan Kuchař (http://www.jankuchar.cz)
+ *
+ * For the full copyright and license information, please view
+ * the file license.txt that was distributed with this source code.
+ */
+
+namespace MultipleFileUpload\Model\MySQL;
+
+use MultipleFileUpload\Model\BaseQueue,
+	Nette\Environment,
+	Nette\Http\FileUpload,
+	Nette\InvalidStateException,
+	SQLite3,
+	SQLite3Result,
+	SQLite3Stmt;
+
+class Queue extends BaseQueue
+{
+
+    /**
+	 * Execute query
+	 * @param string $sql
+	 * @return SQLite3Result
+	 */
+	function query($sql)
+	{
+		return $this->getQueuesModel()->query($sql);
+	}
+
+	/**
+	 * Add file to queue
+	 * @param FileUpload $file
+	 */
+	function addFile(FileUpload $file)
+	{
+		$file->move($this->getUniqueFilePath());
+		$this->getQueuesModel()->getConnection()->query("
+            INSERT INTO files (queueID, created, data, name) VALUES (
+            ?,?,?,?)", $this->getQueueID(), time(), "X'" . bin2hex(serialize($file)) . "'", $file->getName());
+	}
+
+
+	// TODO: rename!!!
+	/**
+	 * Add file to queue manually
+	 * @param FileUpload $file
+	 * @param int $chunk
+	 * @param int $chunks
+	 */
+	function addFileManually($name, $chunk, $chunks)
+	{
+		$this->getQueuesModel()->getConnection()->query("
+		    INSERT INTO files (queueID, created, name, chunk, chunks) VALUES (?, ?, ?, ?, ?)
+        ", $this->getQueueID(), time(), $name, $chunk, $chunks);
+	}
+
+
+	/**
+	 * Update file
+	 * @param string $name
+	 * @param int $chunk
+	 * @param FileUpload $file
+	 */
+	function updateFile($name, $chunk, FileUpload $file)
+	{
+		$this->query("START TRANSACTION");
```

Comment:
As this is solved already in version 1.2.0,, I will remove whole MySQL namespace and skip this.